### PR TITLE
chore: sandwich improvements

### PR DESCRIPTION
**Pull request by petethepig**
Made it do deduplication in callees and callers. Addressed #1659 

Screen Shot 2022-11-02 at 3 21 01 PM

### CODE REVIEW

File: packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts [hunk 1]
```diff
@@ -53,37 +53,40 @@ export const treeToFlamebearer = (tree: TreeNode): FlamebearerData => {
...
     const nextNode = { name, ...rest, total: [total] };
-
     nestedObj.children = [nextNode];
     nestedObj = nextNode;
   });
 
   return result.children[0];
 };
 
+function dedupTree(node) {
```
Comment (Rperry2174) [inline packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:70]: can you add some comments about whats happening here?

Comment (petethepig) [reply packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:67]: It's a recursive function that takes a node as input and then for that node walks over all children and performs deduplication meaning that it finds children with the same name and when two children have the same name it unifies them into 1 child node.

It then recursively calls itself on each new child node.

File: packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts [hunk 2]
```diff
@@ -170,12 +168,15 @@ export function callersFlamebearer(
   processTree(tree);
 
   subtrees.forEach((v, i) => {
-    totalNode.children.push(
-      arrayToTree(v, maxSubtreeLvl, targetFunctionTotals[i])
-    );
+    totalNode.children.push(arrayToTree(v.reverse(), targetFunctionTotals[i]));
```
Comment (petethepig) [inline packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:171]: Major change I made here is that I first make an inverse (upside down) tree.
This allows me to call `dedupTree` function on the whole structure.
And then I reverse it back on line 177.

Comment (petethepig) [reply packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:171]: This allows us to get rid of some other code, e.g all that `maxSubtreeLvl` logic

**Review by dogfrogfog — APPROVED** [review]
LGTM !! :fire:

File: packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts (hunk 1 above)
Comment (eh-am) [reply packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:67]: can we type `node`?

**Review by eh-am — COMMENTED** [review]
Cool! How would one go about adding a test to cover this case? cc @dogfrogfog I think it would be useful.

File: packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts (hunk 2 above)
Comment (eh-am) [reply packages/pyroscope-flamegraph/src/convert/sandwichViewProfiles.ts:171]: i don't think that's obvious, a comment in the code would be appreciated

**Pull request merged**